ui/src/Pages/AlbumPage/AlbumPage.js

```diff
@@ -38,22 +42,70 @@ const albumQuery = gql`
   }
 `
 
+let refetchNeededAll = false
+let refetchNeededFavorites = false
+
 function AlbumPage({ match }) {
   const albumId = match.params.id
+  const [onlyFavorites, setOnlyFavorites] = useState(
+    match.params.subPage === 'favorites'
+  )
+
+  const toggleFavorites = refetch => {
+    const newState = !onlyFavorites
+    if (
+      (refetchNeededAll && !newState) ||
+      (refetchNeededFavorites && newState)
+    ) {
+      refetch({ id: albumId, onlyFavorites: newState }).then(() => {
+        if (onlyFavorites) {
+          refetchNeededFavorites = false
+        } else {
+          refetchNeededAll = false
+        }
+        setOnlyFavorites(newState)
+      })
+    } else {
+      setOnlyFavorites(newState)
+    }
+    history.replaceState(
+      {},
+      '',
+      '/album/' + albumId + (onlyFavorites ? '/favorites' : '')
```

Comment:
When navigating to a `/album/:id` page, and click show only favorites, everything works as expected, except that the url doesn't update to `/album/:id/favorite`, but when you click it again to show all images again the url updates to `/album/:id/favorite`.

I could imagine it's because `onlyFavorites` on this line should be changed to `newState`

Reply:
Yeah, I forgot this. I did a refactoring and somehow miss that.